File: java/sprykerFileCreator/model/manager/ClassManagerInterface.java
package sprykerFileCreator.model.manager;

import com.intellij.psi.PsiDirectory;
import com.intellij.psi.PsiElement;

import java.util.HashMap;

public interface ClassManagerInterface {

    PsiElement handleClass(PsiDirectory fileDirectory, String classType, HashMap<String, String> classConfig) throws Exception;

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/manager/ClassManagerInterface.kt
package com.github.tobi812.sprykerplugin.models.manager

import com.github.tobi812.sprykerplugin.actions.ClassConfig
import com.intellij.psi.PsiDirectory
import com.intellij.psi.PsiElement
import java.lang.Exception

interface ClassManagerInterface {
    @Throws(Exception::class)
    fun handleClass(
        fileDirectory: PsiDirectory,
        classType: String,
        classConfig: ClassConfig
    ): PsiElement
}